lib/api/model/model.dart

```diff
@@ -604,6 +613,37 @@ extension type const TopicName(String _value) {
   /// using [canonicalize].
   bool isSameAs(TopicName other) => canonicalize() == other.canonicalize();
 
+  /// Convert this topic to match how it would appear on a message object from
+  /// the server.
+  ///
+  /// This assumes that the topic is originally for a send-message
+  /// request — such a topic must be constructed from a string without
+  /// leading/trailing whitespace.
+  ///
+  /// For a client that does not support empty topics,
+  /// a modern server (FL>=334) would convert "(no topic)" and empty topics to
+  /// `store.realmEmptyTopicDisplayName`.
+  ///
+  /// See also: https://zulip.com/api/send-message#parameter-topic
+  TopicName interpretAsServer({
+    required int zulipFeatureLevel,
+    required String? realmEmptyTopicDisplayName,
+  }) {
+    assert(_value.trim() == _value);
+    // TODO(server-10) simplify this away
+    if (zulipFeatureLevel < 334) {
+      assert(_value.isNotEmpty);
+      return this;
+    }
+
+    if (_value == kNoTopicTopic || _value.isEmpty) {
```

Comment:
The "(no topic)" case here looks like it started more recently than the other, at FL 370:
https://zulip.com/api/changelog